src/test/java/org/assertj/core/api/SoftAssertionsTest.java

```diff
@@ -2608,6 +2608,59 @@ void file_soft_assertions_should_work_with_navigation_methods() {
                                   .containsExactly("[size()] error message", "[content()] error message");
   }
 
+  @Test
+  void file_soft_assertions_should_work_with_binaryContent() {
+    // GIVEN
+    File file = new File("src/test/resources/actual_file.txt");
+    // WHEN
+    softly.assertThat(file)
+      .overridingErrorMessage("error message")
+      .as("size()")
+      .size()
+      .isGreaterThan(0)
+      .isLessThan(1)
+      .returnToFile()
+      .as("binaryContent()")
+      .binaryContent()
+      .startsWith("actual".getBytes(UTF_8))
+      .startsWith("123".getBytes(UTF_8));
+    // THEN
+    then(softly.errorsCollected()).extracting(Throwable::getMessage)
+      .containsExactly("[size()] error message", "[binaryContent()] error message");
+  }
+
+  @Test
+  void path_soft_assertions_should_work_with_content() {
+    // GIVEN
+    Path file = new File("src/test/resources/actual_file.txt").toPath();
```

Comment:
I think you can remove this test as it is similar to `path_soft_assertions_should_report_errors_on_methods_that_switch_the_object_under_test` and rename `path_soft_assertions_should_report_errors_on_methods_that_switch_the_object_under_test` to `path_soft_assertions_should_work_with_content  maybe`  as it is a more accurate test name